crypto/encode_decode/decoder_meth.c

```diff
@@ -315,15 +315,15 @@ static OSSL_DECODER *inner_ossl_decoder_fetch(OSSL_LIB_CTX *libctx, int id,
     }
 
     /*
-     * If we have been passed neither a name_id or a name, we have an
+     * If we have been passed both a name_id or a name, we have an
      * internal programming error.
      */
-    if (!ossl_assert(id != 0 || name != NULL)) {
+    if (!ossl_assert(id == 0 || name == NULL)) {
```

Comment:
Would ^ be better here?

Reply:
Uhm, why?

Reply:
I must be misunderstanding.. Is this the case?
id = 0 and name = NULL =>invalid
id = valid and name = NULL =>valid
id = valid and name = valid => valid
id = valid and name = valid  = > invalid

Reply:
Not quite.  This is the case:

id = 0 and name = NULL => valid
id = valid and name = NULL =>valid
id = valid and name = valid => valid
id = valid and name = valid = > invalid

Reply:
id = 0 and name = NULL => is that a fetch all..

Reply:
That's a pre-fetch, i.e. what you end up having when doing `OSSL_DECODER_fetch(libctx, NULL, propq)`.